recipes/imagl/all/conanfile.py

```diff
@@ -44,27 +73,95 @@ def source(self):
+                toolset_version = tools.vcvars_dict(self).get("VCToolsVersion", os.environ.get("VCToolsVersion",""))
+            except ConanException:
+                raise ConanInvalidConfiguration("Your Visual Studio compiler seems not to be installed in a common way. It is not supported. This unfortunately happens in conan center index. To build imaGL, append '--build missing' to your 'conan install' command line.")
+
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("imagl requires C++20, which your compiler does not fully support.")
+        minimum_toolset_version = self._compiler_toolsets_minimum_version.get(str(self.settings.compiler), "0")
+        if not minimum_version:
+            self.output.warn("imaGL requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+        elif lazy_lt_semver(compiler_version, minimum_version) or lazy_lt_semver(toolset_version, minimum_toolset_version):
+            raise ConanInvalidConfiguration("imaGL requires some C++20 features, which your {} {} compiler does not support.".format(str(self.settings.compiler), compiler_version)
+                                         + (" Your Visual Studio toolset version is {}; version {}+ is required.".format(toolset_version, str(minimum_toolset_version)) if minimum_toolset_version else ""))
+        elif str(self.settings.compiler) == "clang" and compiler_version == "11" and not self.options.allow_clang_11:
+            raise ConanInvalidConfiguration("Clang 11 is not currently supported by conan center index. To build imaGL, append '-o imagl:allow_clang_11=True --build missing' to your 'conan install' command line.")
         else:
-            self.output.warn("imagl requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+            print("Your compiler is {} {} and is compatible.".format(str(self.settings.compiler), compiler_version))
+            if minimum_toolset_version != "0":
+                print("Its toolset is in version {} and is compatible. Minimum version is {}.".format(toolset_version, minimum_toolset_version))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._supports_jpeg:
+            del self.options.with_jpeg
+        if not str(self.settings.compiler) == "clang" or not str(self.settings.compiler.version) == "11":
+            del self.options.allow_clang_11
+        else:
+            self.output.warn("allow_clang_11 option will be removed in the future when conan center index will support clang 11.")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.settings.compiler.cppstd:
+            self.settings.compiler.cppstd = "20"
```

Comment:
Do not assign a value here, it is not going to work... and future versions of Conan will fail here.